### DISCUSSION TITLE
Unit go-agent.service could not be found.

### DISCUSSION THREAD

**Discussion by welloncn** [gocd/gocd#12700] [Q&A]
![image](https://github.com/gocd/gocd/assets/4989505/64c096c6-bbe5-46ea-89e3-81cf21a6c6ae)
安装后无法找到这个服务
使用deb安装也不行

执行的操作如下：
```
sudo install -m 0755 -d /etc/apt/keyrings
curl https://download.gocd.org/GOCD-GPG-KEY.asc | sudo gpg --dearmor -o /etc/apt/keyrings/gocd.gpg
sudo chmod a+r /etc/apt/keyrings/gocd.gpg
echo "deb [signed-by=/etc/apt/keyrings/gocd.gpg] https://download.gocd.org /" | sudo tee /etc/apt/sources.list.d/gocd.list
sudo apt-get update
sudo apt-get install go-agent
```

**Comment by chadlwilson** [comment]
I don't understand Chinese so you're probably going to have to translate to English to explain what you are trying to do, which specific distribution you are trying to install on and which init system you are trying to use.

If I recall correctly, the installer/`.deb` tries to detect the init system and install as appropriate. It should normally log something on package install such as the below:

```
Setting up go-agent (23.5.0-18179) ...
Detected Linux:
Installing the go-agent daemon with init.d (update-rc.d)...
Installation of GoCD Agent completed.
Now please edit /usr/share/go-agent/wrapper-config/wrapper-properties.conf and set the server to the url address of your GoCD Server (http://example.com:8153/go).
To ensure full end-to-end transport security between the agent and server
see https://docs.gocd.org/current/installation/ssl_tls/end_to_end_transport_security.html
Once that is done start the GoCD Agent with '/etc/init.d/go-agent start'
```

I don't see anything like that in your log, instead there are errors like "Must be root to perform this action" which looks like a problem. If you see binaries within `/usr/share/go-agent` then the agent has installed, it just possibly has not installed a service unit correctly?

If you are already installing using user `root`, you don't need to `sudo` - so I am wondering if it behaves differently if you don't sudo and just `apt-get reinstall go-agent` instead?

**Reply by welloncn** [reply]
yes, but /etc/init.d/go-agent not found
![image](https://github.com/gocd/gocd/assets/4989505/54eac511-98ff-47af-b10d-d913d5dd5695)
![image](https://github.com/gocd/gocd/assets/4989505/2c07a32e-184f-4d9b-b08f-890e3a4dbfe3)

**Reply by chadlwilson** [reply]
Please answer the other questions, or I can't really help. The 'must be root' error is probably related but I need to be able to replicate your problem as it works fine for me on Debian 10/11/12.
